Automattic/jetpack
# Google Fonts: enqueue only picked font families in Gutenberg

#### Changes proposed in this Pull Request:

As https://github.com/WordPress/gutenberg/pull/39559 got merged, we have more control over which font faces get to the front-end after registering.

This PR introduces code that introspects block usage and global styles configuration looking for Google fonts that have been registered.

#### Jetpack product discussion


N/A.

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






Setting up the environment:

1. Run `trunk` Gutenberg;
2. Map Gutenberg to Jetpack docker volume (doing this instead of `a8c-wp-env` because we had loads of trouble trying to get it working);
3. Run `composer install` inside `projects/plugins/jetpack` so the module gets linked with the plugin;
4. Enable the Google fonts module in Jetpack settings;

Enqueueing fonts used in blocks:

1. Open the site editor;
2. Include a block that allows typography customization;
3. Pick one of the registered Google fonts (e.g. Roboto);
4. Save the changes;
5. Open the site.

In the page source, check that only the picked font family shows up in the `style#webfonts-inline-css` element.

Enqueueing fonts used in global styles:

1. Open the site editor;
2. Open the global styles settings;
3. Pick one of the registered Google fonts (e.g. Roboto) in the typography section;
4. Save the changes;
5. Open the site.

In the page source, check that only the picked font families, both used in blocks above and the one that you just set in global styles, shows up in the `style#webfonts-inline-css` element.
